packages/ssi-types/src/types/vc.ts

```diff
@@ -90,57 +95,90 @@ export interface WrappedVerifiableCredential {
   /**
    * Original VC that we've received
    */
-  original: string | JwtWrappedVerifiableCredential | IVerifiableCredential
+  original: OriginalVerifiableCredential
   /**
    * In case of JWT credential it will be the decoded version. In other cases it will be the same as original one
    */
-  decoded: JwtWrappedVerifiableCredential | IVerifiableCredential
+  decoded: JwtDecodedVerifiableCredential | IVerifiableCredential
   /**
-   * Type of this credential. Supported types are json-ld and jwt
+   * Type of this credential. Supported types are json-ld and jwt (decoded/encoded)
    */
-  type: VerifiableDataExchangeType
+  type: OriginalType
   /**
-   * created based on https://www.w3.org/TR/vc-data-model/#jwt-decoding
+   * The claim format, typically used during exchange transport protocols
    */
-  internalCredential: ICredential
+  format: ClaimFormat
+  /**
+   * Internal stable representation of a Credential without Proofs, created based on https://www.w3.org/TR/vc-data-model/#jwt-decoding
+   */
+  credential: ICredential
 }
 
 export interface WrappedVerifiablePresentation {
-  original: string | JwtWrappedVerifiablePresentation | IVerifiablePresentation
-  decoded: JwtWrappedVerifiablePresentation | IVerifiablePresentation
-  type: VerifiableDataExchangeType
-  internalPresentation: InternalPresentation
+  /**
+   * Original VP that we've received
+   */
+  original: OriginalVerifiablePresentation
+  /**
+   * In case of JWT VP it will be the decoded version. In other cases it will be the same as original one
+   */
+  decoded: JwtDecodedVerifiablePresentation | IVerifiablePresentation
+  /**
+   * Type of this Presentation. Supported types are json-ld and jwt (decoded/encoded)
+   */
+  type: OriginalType
+  /**
+   * The claim format, typically used during exchange transport protocols
+   */
+  format: ClaimFormat
+  /**
+   * Internal stable representation of a Presentation without proofs, created based on https://www.w3.org/TR/vc-data-model/#jwt-decoding
+   */
+  presentation: UniformVerifiablePresentation
+  /**
+   * Wrapped Verifiable Credentials belonging to the Presentation
+   */
   vcs: WrappedVerifiableCredential[]
 }
 
-export enum VerifiableDataExchangeType {
-  JSONLD,
-  JWT_ENCODED,
-  JWT_DECODED,
+export enum OriginalType {
```

Comment:
suggest to revert this one. `OriginalType` is too general

Reply:
Problem is that it can be used for Credentials and Presentations and is scoped to a Wrapped VC or Wrapped VP, which also has an original field

I don't really have a better name tbh especially since it is clear from the context